draft-ietf-mls-protocol.md

```diff
@@ -566,18 +566,22 @@ ReadVarint(data):
   // first two bits of the first byte.
   v = data.next_byte()
   prefix = v >> 6
+  if prefix == 3:
+    raise Exception('invalid variable length integer prefix')
+
   length = 1 << prefix
 
   // Once the length is known, remove these bits and read any
   // remaining bytes.
   v = v & 0x3f
   repeat length-1 times:
     v = (v << 8) + data.next_byte()
-  return v
 
   // Check that the encoder used the minimum bits required
-  if length > 1 && v < (1 << (length - 1)):
-    raise an exception
+  if length > 1 && v < (1 << ((8*(length-1))-2)):
```

Comment:
The fix is still buggy, here is the correct one:
```suggestion
  if prefix >= 1 && v < (1 << (8*(1 << (prefix-1))-2)):
```

Reply:
What's the bug in the current code?

Reply:
In your code, all numbers between `2^14` and `2^22` will be rejected: they are represented with `prefix = 2` (hence `length = 4`). In this case, the buggy check is `v < (1 << (8*3-2))`, the fixed check is `v < (1 << (8*2-2))`.

Reply:
Ah, thank you!